extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -0,0 +1,325 @@
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errors += " Parameter 'apikey' must be configured.";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.location) == null) {
+            errors += " Parameter 'location' must be configured.";
+            validConfig = false;
+        }
+        if (config.language != null) {
+            String lang = StringUtils.trimToEmpty(config.language);
+            if (lang.length() != 2) {
+                errors += " Parameter 'language' must be 2 letters.";
+                validConfig = false;
+            }
+        }
+        if (config.refresh != null && config.refresh < 5) {
+            errors += " Parameter 'refresh' must be at least 5 minutes.";
+            validConfig = false;
+        }
+        errors = errors.trim();
+
+        if (validConfig) {
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+        } else {
+            logger.info("Disabling thing '{}': {}", getThing().getUID(), errors);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errors);
+        }
+    }
+
+    /**
+     * Start the job refreshing the weather data
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request new weather data to the Weather Underground service
+                        updateWeatherData();
+
+                        // Update all channels from the updated weather data
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId());
+                        }
+                    } catch (Exception e) {
+                        logger.debug("Exception occurred during execution: {}", e.getMessage());
+                    }
+                }
+            };
+
+            WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
+            int period = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleAtFixedRate(runnable, 0, period, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing WeatherUnderground handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId());
+        } else {
+            logger.info("The Weather Underground binding is a read-only binding and cannot handle command {}", command);
```

Comment:
Reduce to debug as maybe another binding is linked to the same item and can handle the command - so your message might be noise.